gstreamer-sdp/src/s_d_p_media.rs

```diff
@@ -0,0 +1,448 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+// //
+// // Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// // http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// // option. This file may not be copied, modified, or distributed
+// // except according to those terms.
+
+use std::mem;
+use std::ptr;
+
+use ffi;
+use glib::translate::*;
+use glib_ffi;
+use gst;
+
+use auto::SDPResult;
+use m_i_k_e_y_message::MIKEYMessage;
+use s_d_p_attribute::SDPAttribute;
+use s_d_p_bandwidth::SDPBandwidth;
+use s_d_p_connection::SDPConnection;
+use s_d_p_key::SDPKey;
+
+glib_wrapper! {
+    pub struct SDPMedia(Boxed<ffi::GstSDPMedia>);
+
+    match fn {
+        copy => |ptr| ffi::gst_sdp_media_copy(mut_override(ptr)),
+        free => |ptr| ffi::gst_sdp_media_free(ptr),
+    }
+}
+
+impl SDPMedia {
+    pub fn add_attribute<'a, P: Into<Option<&'a str>>>(&mut self, key: &str, value: P) -> Result<_, ()> {
```

Comment:
You need to provide a type for the successful case. This should be `Result<(), ()>` I guess